dotnet/samples/Demos/VectorStoreRAG/TextSnippet.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Data;
+
+namespace VectorStoreRAG;
+
+/// <summary>
+/// Data model for storing a section of text with an embedding and an optional reference link.
+/// </summary>
+/// <typeparam name="TKey">The type of the data model key.</typeparam>
+internal sealed class TextSnippet<TKey>
```

Comment:
A "Name" property is useful. If you ask the LLM to generate citations it will typically create something like this: "[Name]<Link>"

Reply:
I don't really have anything to put into it from the PDF though. Do you have anything in mind?

Reply:
You could add something like `<PDF filename>#page=<page number>`

The model can then construct citations like this: 
`[semantic-kernel.pdf#page=35](file:///C:/Users/me/Documents/semantic-kernel.pdf#page=35)`

Reply:
Thanks, fixed.